engine/execution/computation/computer/computer.go

```diff
@@ -425,6 +476,11 @@ func (e *blockComputer) executeTransactionInternal(
 		return txn, err
 	}
 
+	err = txn.Validate()
```

Comment:
why validating multiple times?

Reply:
i've added some comments to clarify this.